polkadot/runtime/parachains/src/assigner_on_demand/mod.rs

```diff
@@ -790,6 +636,35 @@ where
 		})
 	}
 
+	/// Collect the revenue from the `when` blockheight
+	pub fn claim_revenue_until(when: BlockNumberFor<T>) -> BalanceOf<T> {
+		let now = <frame_system::Pallet<T>>::block_number();
+		let mut amount: BalanceOf<T> = BalanceOf::<T>::zero();
+		Revenue::<T>::mutate(|revenue| {
+			while !revenue.is_empty() {
+				let index = (revenue.len() - 1) as u32;
+				if when > now.saturating_sub(index.into()) {
+					amount = amount.saturating_add(revenue.pop().defensive_unwrap_or(0u32.into()));
+				} else {
+					break
+				}
+			}
+		});
+
+		let imbalance = T::Currency::burn(amount);
+		let _ = T::Currency::settle(
+			&Self::account_id(),
+			imbalance,
+			WithdrawReasons::FEE,
+			ExistenceRequirement::AllowDeath,
+		);
+		amount
```

Comment:
`burn` will handle total issuance, but I think it depends on intent. As in, if we want all the burning to be done at once from the Coretime Chain (like, "look how much we burned from Coretime sales") then teleporting probably makes sense and having Coretime manage all the revenue burning. cc @muharem

Reply:
This is a combination of system and private income. The private income is an incentive for people to pool cores, so it is distributed to the people who contributed to the pool by the coretime chain before the remaining system amount is dealt with in the same way as bulk revenue.

Reply:
Yeah I discussed a bit with @s0me0ne-unkn0wn in DM. Since these get minted on the Coretime chain, we should definitely use `teleport` and let Coretime handle what to do with revenue (which may mean teleporting it back to burn).